includes/scheduler/class-comment.php

```diff
@@ -81,6 +86,54 @@ public static function schedule_comment_activity( $new_status, $old_status, $com
 		add_to_outbox( $comment, $type, $comment->user_id );
 	}
 
+	/**
+	 * Announce an interaction.
+	 *
+	 * When a comment is received from another ActivityPub instance and approved,
+	 * this method creates an Announce activity so the blog's followers are notified
+	 * about the interaction.
+	 *
+	 * @param string      $new_status The new comment status.
+	 * @param string      $old_status The old comment status.
+	 * @param \WP_Comment $comment    The comment object.
+	 */
+	public static function maybe_announce_interaction( $new_status, $old_status, $comment ) {
+		// Only if we're in both Blog and User modes.
+		if ( ACTIVITYPUB_ACTOR_AND_BLOG_MODE !== \get_option( 'activitypub_actor_mode', ACTIVITYPUB_ACTOR_MODE ) ) {
+			return;
+		}
+
+		if ( 'approved' !== $new_status || 'approved' === $old_status ) {
+			return;
+		}
+
+		if ( ! Comment_Utils::was_received( $comment ) ) {
+			return;
+		}
+
+		// Get activity from comment meta.
+		$activity = \get_comment_meta( $comment->comment_ID, '_activitypub_activity', true );
```

Comment:
Could we get this from the Inbox, instead of saving it in comment meta?
Alternatively, could we use the `source_id` meta as the object instead of offering the full object?

Reply:
We can postpone this a bit until we have the inbox activated by default. PR incoming.